src/sa_pool.c

```diff
@@ -506,31 +506,29 @@ int sa_fetch(struct netif_port *dev, const struct sockaddr_in *daddr,
     int err;
     assert(saddr);
 
-    if (saddr->sin_addr.s_addr != INADDR_ANY && saddr->sin_port != 0)
+    if (saddr &&  saddr->sin_addr.s_addr && saddr->sin_addr.s_addr != INADDR_ANY && saddr->sin_port != 0)
         return 0; /* everything is known, why call this function ? */
 
     /* if source IP is assiged, we can find ifa->this_sa_pool
      * without @daddr and @dev. */
-    if (saddr && saddr->sin_addr.s_addr) {
-        ifa = inet_addr_ifa_get(AF_INET, dev,
-                                (union inet_addr*)&saddr->sin_addr);
-        if (!ifa)
-            return EDPVS_NOTEXIST;
-
-        if (!ifa->this_sa_pool) {
-            RTE_LOG(WARNING, SAPOOL, "%s: fetch addr on IP without pool.",
-                    __func__);
-            inet_addr_ifa_put(ifa);
-            return EDPVS_INVAL;
-        }
+    ifa = inet_addr_ifa_get(AF_INET, dev,
```

Comment:
`if (saddr->sin_addr.s_addr)` should be reserved.
This section should be kept not  changed I think, it's for the case only source IP is assigned but not port.